src/systems/isq/include/units/isq/mechanics.h

```diff
@@ -23,50 +23,86 @@
-// quantity_kind?
-// TODO Should we add other forces as well: static_friction_force, kinematic_friction_force, rolling_resistance,
-// drag_force
-DERIVED_DIMENSION(impulse, decltype(force / time));
-DERIVED_DIMENSION(angular_momentum, decltype(length * momentum));  // TODO position_vector
-DERIVED_DIMENSION(moment_of_inertia, decltype(angular_momentum * angular_velocity));
-DERIVED_DIMENSION(moment_of_force, decltype(length * force));  // TODO position_vector
-DERIVED_DIMENSION(torque, decltype(moment_of_force));          // TODO angle?
-DERIVED_DIMENSION(angular_impulse, decltype(moment_of_force * time));
-DERIVED_DIMENSION(pressure, decltype(force / area));
-DERIVED_DIMENSION(stress, decltype(pressure));  // TODO tensor?
-DERIVED_DIMENSION(normal_stress, decltype(force / area));
-DERIVED_DIMENSION(strain, decltype(stress / stress));          // TODO what is a correct equation here?
-DERIVED_DIMENSION(poisson_number, decltype(length / length));  // TODO width?
-// TODO modulus quantities
-DERIVED_DIMENSION(compressibility, decltype(volume / volume / pressure));
-DERIVED_DIMENSION(second_axial_moment_of_area, decltype(area * area));  // TODO what is a correct equation here?
-DERIVED_DIMENSION(section_modulus, decltype(second_axial_moment_of_area / length));  // TODO radial distance
-// TODO friction coefficients?
-DERIVED_DIMENSION(dynamic_viscosity, decltype(stress * length / speed));  // TODO shear stress, velocity
-DERIVED_DIMENSION(kinematic_viscosity, decltype(dynamic_viscosity / mass_density));
-DERIVED_DIMENSION(surface_tension, decltype(force / length));  // TODO what is a correct equation here?
-DERIVED_DIMENSION(power, decltype(force * speed));
-// TODO what about energy (potential and kinetic as separate quantities will prevent an equation for mechanical one, is
-// it expected?)
-DERIVED_DIMENSION(efficiency, decltype(power / power));
-DERIVED_DIMENSION(mass_flow, decltype(mass_density * speed));  // TODO velocity
-DERIVED_DIMENSION(mass_flow_rate, decltype(mass_flow * area));
-DERIVED_DIMENSION(mass_change_rate, decltype(mass / time));
-DERIVED_DIMENSION(volume_flow_rate, decltype(speed * area));  // TODO velocity
-// DERIVED_DIMENSION(action, decltype(energy * time));  // TODO make it compile
+QUANTITY_SPEC(mass_density, mass / volume);
+inline constexpr auto density = mass_density;
+QUANTITY_SPEC(specific_volume, 1 / mass_density);
+QUANTITY_SPEC(relative_mass_density, mass_density / mass_density);
+inline constexpr auto relative_density = relative_mass_density;
+QUANTITY_SPEC(surface_mass_density, mass / area);
+inline constexpr auto surface_density = surface_mass_density;
+QUANTITY_SPEC(linear_mass_density, mass / length);
+inline constexpr auto linear_density = linear_mass_density;
+QUANTITY_SPEC(momentum, mass* velocity);
+QUANTITY_SPEC(force, mass* acceleration);     // vector  // TODO what is a correct equation here?
```

Comment:
> `// TODO what is a correct equation here?`

Many quantities are described without one in ISO 80000. There's also those whose equation are inexpressible in C++ with just the operators. You should be able to specify just the dimension instead.

```C++
inline constexpr auto force                   = defn<"𝙁", "dim 𝙁 = 𝘮𝘭𝘵⁻²", mass("𝘮"), length("𝘭"), time("𝘵")>();
```

Reply:
Instead, I think we should introduce those operations explicitly. I spent yesterday evening figuring out how to make a cross-product work. Writing a function dealing with the objects is easy (I did it yesterday already). I still have to figure out how to express that in the expression template of `derived_quantity_spec` though.